src/pivot-table/hooks/__tests__/use-sorting.test.ts

```diff
@@ -0,0 +1,206 @@
+/* eslint-disable @typescript-eslint/no-unused-expressions, @typescript-eslint/unbound-method, jest/valid-expect, @typescript-eslint/return-await */
```

Comment:
These 4 es lint rule gets more annoying rather than being helpful for:
```
expect(async () => await changeSortOrder(header, newSortDirection)).toThrow();
```

basically i'm expecting that function to throw an error in my test, If i do it in try...catch like:
```
try {
  await changeSortOrder(header, newSortDirection);
} catch (err: any) {
  expect(err.message).toBe("No Model provided!");
}
```
I'm getting another lint error!
<img width="550" alt="Screenshot 2023-09-01 at 15 36 28" src="https://github.com/qlik-oss/sn-pivot-table/assets/29652890/409f0d3a-f69e-4318-9ab7-d5d6585da759">

Reply:
Use type [unknown](https://mariusschulz.com/blog/the-unknown-type-in-typescript) instead of any

Put the err.message in a variable that is defined outside of the catch block and do the expect at the end

Something like this
`
let msg = "";
try {
  await changeSortOrder(header, newSortDirection);
} catch (err: unkonwn) {
  msg = err.message;
}
expect(msg).toBe("No Model provided!");
`